src/conventions/h-group/clue-interpretation/focus-possible.js

```diff
@@ -41,17 +41,19 @@ function find_colour_focus(game, suitIndex, action) {
 	const old_play_stacks = state.play_stacks;
 	let already_connected = [focused_card.order];
 
+	let firstPlay = true;
 	let finesses = 0;
+	const promised = [];
 
 	while (next_rank < state.max_ranks[suitIndex]) {
 		const identity = { suitIndex, rank: next_rank };
 		state.play_stacks = old_play_stacks.slice();
 
 		// Note that a colour clue always looks direct
-		const ignoreOrders = game.next_ignore[next_rank - old_play_stacks[suitIndex] - 1]?.filter(i =>
-			i.inference === undefined || i.inference.suitIndex === suitIndex).map(i => i.order);
+		const ignoreOrders = game.next_ignore[next_rank - old_play_stacks[suitIndex] - 1]?.map(i => i.order);
```

Comment:
Is there a reason why the extra filter was removed? I think this was to prevent incorrectly ignoring cards when the inference of the focused card is different from the case that's different from the one being rewinded, even on the same card and at the same connection index. I might not have written a test case for it though, I'll see if I can recall the conditions and whether they're necessary.

Reply:
Yes, this was failing to ignore bluffed cards resulting in it seeing the same bluff connection when the bluff wasn't played into on a rewind and replay. There are tests covering this, I don't remember which one off the top of my head but can check.

Reply:
Hmm ... I'm not sure anymore. I know I did this to fix one of the earlier level tests that was failing but perhaps one of the other things I fixed resulted in that test working. I've restored the filter to the way it was before.

Reply:
I suspect this change fixed the issue:

https://github.com/WillFlame14/hanabi-bot/pull/184/files#diff-139b83de75bd680f692b861b2b2cda2484860633d9397623d1de63322647aa86R75

As it results in it no longer thinking the bluff in its own hand in "someone else having the connecting card" in the replay. This was necessary for the test `plays queued finesses in the right order` which expects that it knows after the b1 plays that its next card is connecting.